riptide-core/src/main/java/org/zalando/riptide/DefaultMessageReader.java

```diff
@@ -44,7 +44,12 @@
     private <I> I readBody(final Type type, final ClientHttpResponse response) throws IOException {
         final ResponseExtractor<I> extractor = new HttpMessageConverterExtractor<>(type, converters);
         try {
-            return extractor.extractData(response);
+            try {
+                return extractor.extractData(response);
+            } catch (final IOException | RuntimeException e) {
+                response.close();
+                throw e;
+            }
         } catch (final RestClientException e) {
             propagateIfPossible(e.getCause(), IOException.class, HttpMessageNotReadableException.class);
```

Comment:
Now that we figured out why we have this statement, maybe we should put a comment on it. At least for our future selves.